src/pages/CourseDetailsPage.tsx

```diff
@@ -0,0 +1,114 @@
+import {useEffect, useRef, useState} from 'react';
+import {useParams, useSearchParams} from 'react-router-dom';
+import {Grid, Typography} from '@mui/material';
+import DoneIcon from '@mui/icons-material/Done';
+
+import {PageLoadStages} from '../components/common/PageLoadStages';
+import {PageLayout} from '../components/common/PageLayout';
+import {BaseList} from '../components/common/BaseList';
+import {LessonsList} from '../components/LessonsList';
+import {LessonCard} from '../components/LessonCard';
+import {Rating} from '../components/common/Rating';
+
+import {storageService} from '../service/storage.service';
+import {courseService} from '../service/course.service';
+import {dateFormatter} from '../helpers/dateFormatter';
+import {useFetch} from '../hooks/useFetch';
+import {CourseDetails, Lesson} from '../../types';
+
+
+export const CourseDetailsPage = () => {
+    const {courseId} = useParams();
+    const [searchParams, setSearchParams] = useSearchParams();
+    const videoRef = useRef<HTMLVideoElement>(null);
+
+    const {
+        loading,
+        data,
+        error
+    } = useFetch<CourseDetails>(courseService.getCourse.bind(null, `${courseId}`));
+    const [currentLesson, setCurrentLesson] = useState<Lesson | null>(null);
+
+    const handlerChangeLesson = (id: string) => {
+        const lesson = data!.lessons.find(el => el.id === id);
+        if (!lesson) {
+            return;
+        }
+
+        storageService.setItem(`${courseId}`, id);
+        setCurrentLesson(lesson);
+        setSearchParams({lesson: id});
+    };
+
+    useEffect(() => {
+        if (data) {
+            data.lessons.sort((a, b) => a.order - b.order);
+            const lessonId = searchParams.get('lesson') || storageService.getItem(data.id);
+            const lesson = data.lessons.find(el => el.id === lessonId);
+            setCurrentLesson(lesson ?? data.lessons[0]);
+        }
+    }, [data, searchParams]);
+
+    useEffect(() => {
+        const ref = videoRef.current;
+
+        if (currentLesson) {
+            const currentVideoTime = Number(storageService.getItem(currentLesson.id));
+            ref && currentVideoTime && (ref.currentTime = currentVideoTime);
```

Comment:
**ref.currentTime = currentVideoTime** is always true.
Maybe **ref.currentTime === currentVideoTime**?